yang/frr-zebra.yang

```diff
@@ -0,0 +1,1887 @@
+      }
+      enum "disabled" {
+        value 1;
+        description
+          "Flooding disabled.";
+      }
+      enum pim-sm {
+        value 2;
+        description
+          "Multicast PIM-SM.";
+      }
+    }
+  }
+
+  /*
+   * Information about EVPN VNIs
+   */
+  grouping vni-information {
+      choice type-choice {
+        case l2 {
+          leaf is-layer2 {
+            type empty;
+            description
+              "Information about an L2 VNI.";
+          }
+          leaf vtep-count {
+            type uint32;
+            description
+              "Number of VTEPs.";
+          }
+        }
+        case l3 {
+          leaf is-layer3 {
+            type empty;
+            description
+              "Information about an L3 VNI.";
+          }
+        }
+      }
+      leaf vni-id {
+        type vni-id-type;
+        description
+          "The VNI identifier.";
+      }
+      leaf vxlan-ifname {
+        type string;
+        description
+          "The VxLAN interface name.";
+      }
+      leaf mac-count {
+        type uint32;
+        description
+          "Number of valid MACs.";
+      }
+      leaf neighbor-count {
+        type uint32;
+        description
+          "Number of neighbors.";
+      }
+      leaf vrf {
+        type string;
+        description
+          "The tenant VRF.";
+      }
+      leaf local-vtep-addr {
+        type inet:ipv4-address;
+        description
+          "The local VTEP IP address.";
+      }
+  }
+
+  /*
+   * Detailed EVPN VNI information for L2.
+   */
+  grouping vni-l2-detail {
+    leaf if-index {
+      type uint32;
+      description
+        "The VxLAN ifindex.";
+    }
+    leaf advertise-gw {
+      type empty;
+      description
+        "The gateway MAC-IP is being advertised.";
+    }
+    leaf mcase-group {
+      type rt-types:ipv4-multicast-group-address;
+      description
+        "The VNI multicast group for BUM traffic.";
+    }
+    list remote-vtep-list {
+      leaf remote-vtep {
+        type inet:ipv4-address;
+        description
+          "The remote VTEP IP address.";
+      }
+      leaf vtep-flood {
+        type vni-vtep-flood-type;
+      }
+    }
+  }
+
+  /*
+   * Detailed EVPN VNI information for L3.
+   */
+  grouping vni-l3-detail {
+    leaf svi-interface {
+      type string;
+      description
+        "The SVI interface.";
+    }
+    leaf is-up {
+      type empty;
+      description
+        "The state is active.";
+    }
+    leaf prefix-only {
+      type empty;
+      description
+        "Prefix routes only";
+    }
+    leaf router-mac {
+      type yang:mac-address;
+      description
+        "The VNI Router MAC address.";
+    }
+    list vni-list {
+      description
+        "A list of the associated L2 VNIs.";
+      leaf vni-id {
+        type vni-id-type;
+        description
+          "An L2 VNI identifier.";
+      }
+    }
+  }
+
+  /*
+   * Debug options
+   */
+  grouping zebra-debugs {
```

Comment:
I would prefer excluding this for now & doing something more generic for debug flags overall.

Reply:
I can't quite figure out how to act on that comment? I need to be able to offer support to configure (and retrieve) zebra's debugs, and those are pretty zebra-specific. These are just booleans: what's the benefit in "something ... generic" ?

Reply:
I see where you're coming from @mjstapp but we do need to think about this more. All daemons have debugs and we need to decide how to model them properly before we commit to a particular model in one daemon.